index.js

```diff
@@ -145,16 +145,22 @@ function $asInteger (i) {
 }
 
 function $asNumber (i) {
-  var num = Number(i)
-  if (isNaN(num)) {
-    return 'null'
-  } else {
-    return '' + num
+  if (i === 0) return '0'
+  if (i !== '' && i !== null && i !== undefined && i !== true && i !== false && (i.constructor === Number || typeof i !== 'object')) {
```

Comment:
Can you get away with:

```
const type = typeof i
if (type === 'number' && isFinite(i)) {
  return '' + i
} else if (typeof === 'string') {
  const num = Number(i)
  if (isFinite(num)) {
    return '' + num
  }
  throw new Error()
}
```

Reply:
@allevo have you seen this?

Reply:
That implementation allow the empty string `''` to be coerced to `0`.

The benchmark:
```
$asNumber x 62,418,169 ops/sec ±2.23% (75 runs sampled)
$asNumberOld x 92,619,961 ops/sec ±2.14% (78 runs sampled)
```

As you can see that implementation doesn't impact the speed.

As already said, the problem here is the `throw` statement.